src/libstd/sys_common/backtrace.rs

```diff
@@ -127,13 +127,27 @@ fn filter_frames(frames: &[Frame],
     (skipped_before, skipped_after)
 }
 
+static MARK_START: fn(&mut FnMut()) = mark_start;
 
-/// Fixed frame used to clean the backtrace with `RUST_BACKTRACE=1`.
+/// Fixed frame used to clean the backtrace with `RUST_BACKTRACE=1`
 #[inline(never)]
-pub fn __rust_begin_short_backtrace<F, T>(f: F) -> T
-    where F: FnOnce() -> T, F: Send, T: Send
-{
-    f()
+fn mark_start(f: &mut FnMut()) {
+    f();
+    unsafe {
+        asm!("" ::: "memory" : "volatile"); // A dummy statement to prevent tail call optimization
+    }
+}
+
+/// Convenience wrapper for `mark_start`
+#[unstable(feature = "rt", reason = "this is only exported for use in libtest", issue = "0")]
```

Comment:
Could this `reason` be updated as I think this is used by threading/main now, right?

Reply:
Those are private usages in `libstd` though, which do not require an export.